### PR TITLE
fix: ensure legacy runtime config types are populated

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Context: https://github.com/nuxt/nuxt/pull/28924, https://github.com/nuxt/ecosystem-ci/actions/runs/10799722624.

Currently changes to `NitroRuntimeConfig` and  `NitroRuntimeConfigApp` are breaking for users/modules consuming `nitropack`.

This ensures types imported from `nitropack` aren't breaking.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/core/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import type { NitroConfig } from "nitropack/types";
+import type { NitroConfig, NitroRouteConfig } from "nitropack/types";
+import type { NitroOpenAPIConfig } from "../types/openapi";
```
Comment (pi0) [inline src/core/index.ts:2]: Can we use nitropack/types entry to import this?

Comment (danielroe) [reply src/core/index.ts:2]: It wasn't exported, but if you're happy for it to be, then yes!

Comment (pi0) [reply src/core/index.ts]: At least for v2 lets do it to keep subpath boundaries clear 👍

**Review by pi0 — APPROVED** [review]
Thank you!🙏

**Pull request merged**